src/blocks/table/Table.tsx

```diff
@@ -45,7 +54,64 @@ const StyledRowCell = styled(Cell)<{ cellAlignment?: Column['cellAlignment'] }>`
       : ''}
 `;
 
-const Table: FC<TableProps> = ({ backgroundColor = 'surface-secondary', columns, dataSource, fixedHeader = false }) => {
+const OverlayContainer = styled.div<{ blur?: boolean }>`
+  position: absolute;
+  z-index: 100;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 100%;
+  height: 100%;
+  flex-direction: column;
+  gap: var(--spacing-xs);
+  opacity: ${({ blur }) => (blur ? '0.5' : '1')};
+`;
+
+const NullStateContainer = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  gap: var(--spacing-xs);
+  flex-direction: column;
+`;
+
+const NullStateTextContainer = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  gap: var(--spacing-xxxs);
+  flex-direction: column;
+`;
+
+const NullStateHeading = styled.span`
+  ${() => getTextVariantStyles('bm-semibold', 'text-primary')}
+`;
+
+const NullStateDescription = styled.span`
+  ${() => getTextVariantStyles('bes-regular', 'text-tertiary')}
+`;
+
+const LoadingText = styled.span`
+  ${() => getTextVariantStyles('bm-semibold', 'text-tertiary')}
```

Comment:
you can keep one text styled component with props to pass the details

Reply:
I did this intentionally
- To keep it simple and 
- To allow me to add additional css to specific text in future.

Reply:
okay